install/kubernetes/istio.yaml

```diff
@@ -1,17 +1,185 @@
 # GENERATED FILE. Use with Kubernetes 1.7+
 # TO UPDATE, modify files in install/kubernetes/templates and run install/updateVersion.sh
```

Comment:
is a follow up PR going to switch to istio.yaml and istio-noauth.yaml ? (as we want auth on by default - but I guess that needs coordination with all the test infra)
curious what is the diff between the 2 files btw - can you paste that 
"diff -u istio.yaml istio-auth.yaml" ?

Reply:
here is the diff if anyone else is interested:
```
--- install/kubernetes/istio.yaml	2017-09-10 12:20:39.000000000 -0700
+++ install/kubernetes/istio-auth.yaml	2017-09-10 12:20:39.000000000 -0700
@@ -553,7 +553,7 @@
 data:
   mesh: |-
     # Uncomment the following line to enable mutual TLS between proxies
-    # authPolicy: MUTUAL_TLS
+    authPolicy: MUTUAL_TLS
     mixerAddress: istio-mixer.istio-system:9091
     discoveryAddress: istio-pilot.istio-system:8080
     ingressService: istio-ingress.istio-system
@@ -788,3 +788,39 @@
           optional: true
 ---
 
+# This CA only issues certs/keys for the namespace it runs in.
+# For demo only. Please use istio-cluster-ca.yaml for real-world use cases.
+# Service account CA runs in.
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: istio-ca-service-account
+  namespace: istio-system
+---
+# Istio CA
+apiVersion: v1
+kind: Deployment
+apiVersion: extensions/v1beta1
+metadata:
+  name: istio-ca
+  namespace: istio-system
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        istio: istio-ca
+    spec:
+      serviceAccountName: istio-ca-service-account
+      containers:
+      - name: istio-ca
+        image: gcr.io/istio-testing/istio-ca:ca328c99ed96a80ca778966d8dfba9c7b0ff6410
+        imagePullPolicy: IfNotPresent
+        env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+---
+
```


to further reduce the difference - is there any harm in having the CA also in -auth less ?

Reply:
authPolicy flag no longer works with pilot from master (check the API change). 
The fact we do not detect this in our testing is a statement about the quality of our tests (https://github.com/istio/pilot/issues/1005).

Reply:
Regarding the latter: I recommend using a consistent label for the generated secrets, so the operator can clean up istio and all produced resources without much hassle.

Reply:
@kyessenov a negative test should be there indeed, if you have one in pilot would you mind putting it in istio/istio too ?
what is the new name of or replacment for `authPolicy: MUTUAL_TLS`
which pilot PR changed that ? (and how come there isn't a followup to change it in istio/istio)

Reply:
I am not changing the authPolicy flag in this PR, what I have is the same as istio/istio.

I followed the istio/pilot#1005, and I see pilot still has it.